lib/services/file/fileservice.core.js

```diff
@@ -514,7 +573,8 @@ FileService.prototype.getShareProperties = function (share, optionsOrCallback, c
 
   var options = extend(true, {}, userOptions);
   var webResource = WebResource.head(share)
-    .withQueryOption(QueryStringConstants.RESTYPE, 'share');
+    .withQueryOption(QueryStringConstants.RESTYPE, 'share')
+    .withQueryOption(QueryStringConstants.SHARE_SNAPSHOT, options.shareSnapshotId);
```

Comment:
No validation to snapshot ID?

Reply:
Didn't add validation for snapshot ID in this change.

We can create another separate CR to add snapshot IDs validation, there are more other previous snapshot ID related methods.